### ISSUE TITLE
Graphs from 2spa

### ISSUE THREAD

**Issue by marklhc** [Gengrui-Zhang/R2spa#23]
It would be helpful to have some regression diagnostics using 2spa. See [this paper](https://www.sciencedirect.com/science/article/abs/pii/S0306460318309596). Some ideas are

- Residual plots
    - [ ] Documentation
- Marginal model plots

**Comment by marklhc** [comment]
To start, we can have a function that has tspa model as input, and output a graph showing diagnostics (e.g., bivariate scatterplots).

**Comment by marklhc** [comment]
TODO:
- [x] Make functions in R package format
- [x] Add smoother to show potential non-linearity (e.g., smoothers)

**Comment by marklhc** [comment]
Base R example: https://drsimonj.svbtle.com/visualising-residuals

**Comment by marklhc** [comment]
- options to adjust title and names?

**Comment by marklhc** [comment]
```
Quitting from lines 128-135 (tspa-vignette.Rmd) 
Error in plot.new() : figure margins too large
Calls: <Anonymous> ... eval -> tspa_plot -> plot -> plot.default -> plot.new
```

**Comment by marklhc** [comment]
Document all arguments

**Comment by Gengrui-Zhang** [comment]
Agenda for next meeting:
- Scatterplots: check out marginal and partial plots 
- Residual plots: focus on marginal plots first, and maybe check out: https://sfcheung.github.io/semdplot/articles/casewise_residuals.html

**Comment by marklhc** [comment]
@Gengrui-Zhang This may be a good reference:

https://www.statmodel.com/examples/webnotes/webnote%2020.pdf

After re-reading the Hallgren et al. paper and the web note, I think for scatter plots, we should

a. Default to using factor scores (or observed variables) from the input data
b. Given the confusion about which trend line to add (e.g., model-implied or data-driven), I think it's best not to add a trend line.
c. Create a function to compute casewise residuals (see [`semdplot`](https://sfcheung.github.io/semdplot/articles/semdplot.html)).

**Comment by marklhc** [comment]
I'm adding a simple example to illustrate the idea of calling a core function in a multiple-group setting

```r
# Core function
my_plot <- function(x, y, title = NULL, ...) {
  plot(x, y, main = title, ...)
}
# Example:
dat <- lavInspect(tspa_fit_3, what = "data")
my_plot(dat[[1]][, "fs_speed"], dat[[1]][, "fs_visual"],
        title = names(dat)[[1]])
# Multiple group with a loop
for (i in seq_along(dat)) {
  my_plot(dat[[i]][, "fs_speed"], dat[[i]][, "fs_visual"],
          title = names(dat)[[i]])
}
# So you can have a combined function
my_tspa_plot <- function(fit) {
  dat <- lavInspect(fit, what = "data")
  if (lavInspect(fit, what = "ngroups") == 1) {
    my_plot(dat[, "fs_speed"], dat[, "fs_visual"],
      title = names(dat)
    )
  } else {
    for (i in seq_along(dat)) {
      my_plot(dat[[i]][, "fs_speed"], dat[[i]][, "fs_visual"],
        title = names(dat)[[i]]
      )
    }
  }
}
my_tspa_plot(tspa_fit_3)
```

**Comment by Gengrui-Zhang** [comment]
@marklhc Hi Mark, thanks so much for your suggestions. I have updated the tspa plot function. It is in a new R script called `tspa_plot_revised`. Now it should support both scatterplot and residual plot. I also ran the CMD check and it gave only one note with no errors and no warnings. I'll continue to work on the casewise residual function and try to address the partial regression plot. Thanks!
